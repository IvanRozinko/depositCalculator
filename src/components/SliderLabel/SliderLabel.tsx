import React from 'react'
import { Box, Typography } from '@material-ui/core';
import { Theme, makeStyles, createStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) => 
  createStyles({
    root: {
      width: 140,
      paddingBottom: 10,
      [theme.breakpoints.down('xs')]: {
        paddingBottom: 30,
        width: 'auto'
      }
    }
  })
);

interface SliderLabelType {
  label: string
};

function SliderLabel({ label }: SliderLabelType) {
  const styles = useStyles();
  return (
    <Box className={styles.root}>
      <Typography component='span'>
        {label}
      </Typography>
    </Box>
  )
}

export default SliderLabel
