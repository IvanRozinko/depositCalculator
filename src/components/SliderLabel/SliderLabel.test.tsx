import SliderLabel from './SliderLabel';
import { shallow } from 'enzyme';

describe('<SliderLabel /> :', () => {
  const label = 'Slider';
  const wrapper = shallow(
    <SliderLabel label={label} />
  )
  
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('label should be displayed', () => {
    expect(wrapper.text()).toBe(label);
  })
})