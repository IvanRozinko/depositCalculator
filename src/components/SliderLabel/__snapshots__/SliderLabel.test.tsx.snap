// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SliderLabel /> : renders correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    component="span"
  >
    Slider
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;

exports[`<SliderLabel /> renders correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    component="span"
  >
    Slider
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;

exports[`<SliderLabel renders correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    component="span"
  >
    Slider
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;

exports[`SliderLabel renders correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    component="span"
  >
    Slider
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;
